lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -407,22 +409,35 @@ putStakeCertificate
     -> ChimericAccount
     -> Put
 putStakeCertificate (PoolId poolId) (ChimericAccount accId) = do
+    assertLength "poolId" poolId 32
+    assertLength "account pubKey" accId 32
```

Comment:
It's actually easy to mess up here, if you don't remember if the keys should be in hex or raw binary when constructing the newtypes.

Only exposing explicitly named smart constructors could be an idea.